apis/filesystem/src/main/java/org/jclouds/filesystem/util/Utils.java

```diff
@@ -49,7 +55,23 @@ public static void deleteRecursively(File file) throws IOException {
             }
          }
       }
-      Files.delete(file.toPath());
+
+      for (int n = 0; n < 10; n++) {
+         try {
+            Files.delete(file.toPath());
+         } catch (DirectoryNotEmptyException dnee) {
+            Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
+            continue;
+         } catch (AccessDeniedException ade) {
+            Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
+            continue;
+         } catch (NoSuchFileException nse) {
+            if (n == 0) throw nse;
+            return;
+         }
+         return;
+      }
+      throw new RuntimeException("Could not delete: " + file.toPath());
```

Comment:
Can we abstract this logic out to a method, including Javadoc explaining the motivations?  Also it seems like we should call this helper in `removeBlob` as well.